source/common/stats/symbol_table_impl.cc

```diff
@@ -250,16 +261,16 @@ void SymbolTable::newSymbol() EXCLUSIVE_LOCKS_REQUIRED(lock_) {
   ASSERT(monotonic_counter_ != 0);
 }
 
-bool SymbolTable::lessThan(const StatName& a, const StatName& b) const {
+bool SymbolTableImpl::lessThan(const StatName& a, const StatName& b) const {
   // Constructing two temp vectors during lessThan is not strictly necessary.
   // If this becomes a performance bottleneck (e.g. during sorting), we could
   // provide an iterator-like interface for incrementally decoding the symbols
   // without allocating memory.
   SymbolVec av = SymbolEncoding::decodeSymbols(a.data(), a.dataSize());
   SymbolVec bv = SymbolEncoding::decodeSymbols(b.data(), b.dataSize());
+  Thread::LockGuard lock(lock_);
```

Comment:
Can you add a comment on what is guarded here? I'm unclear if this shouldn't be even earlier in the method?

Reply:
Added a comment. I don't see why we'd need to hold a lock earlier; no member variables are accessed from the static method. Also updated the thread annotation to make it clearer we need to hold the lock while calling fromSymbol().